### PR TITLE
Observers can no longer use OOC pre-game

### LINKED ISSUE #6675: Cogscarab ghostrole can be used to spoil clock cult in OOC before round start
As the cogscarab ghost role is avaiable as soon as you can select "observe", there is a timeframe of about half a minute between being able to ghost and the round actually starting where OOC is still enabled.
Because of this I have seen several people ghost, see there are cogscarabs and then yell in OOC that its clock cult.

My idea to solve this would be to make the scarabs avaiable only once everyone else spawns, similarly to Ashwalker eggs.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
fixes #6675

## About The Pull Request

Disables OOC for dead mobs before the game starts.

## Why It's Good For The Game

Players that observe during the round setup should not be able to use OOC to give away meta-information about the round. This should be bannable, but also shouldn't be allowed by the code.

## Changelog
:cl:
tweak: Observers can no longer use OOC pre-game.
/:cl:

### CODE REVIEW

**Comment by AnCopper** [comment]
What kind of information could they even provide?

**Comment by PowerfulBacon** [comment]
> What kind of information could they even provide?

Ghost roles including game-mode specific ghost roles such as cogscarabs.

File: code/modules/client/verbs/ooc.dm [hunk 1]
```diff
@@ -16,6 +16,9 @@ GLOBAL_VAR_INIT(normal_ooc_colour, "#002eb8")
 		if(!GLOB.ooc_allowed)
 			to_chat(src, "<span class='danger'>OOC is globally muted.</span>")
 			return
+		if(mob.stat == DEAD && SSticker.current_state < GAME_STATE_PLAYING)
```
Comment (Archanial) [inline code/modules/client/verbs/ooc.dm]: you are checking stat == dead twice now

Comment (PowerfulBacon) [reply code/modules/client/verbs/ooc.dm]: oh yea

File: code/modules/client/verbs/ooc.dm [hunk 2]
```diff
@@ -16,6 +16,9 @@ GLOBAL_VAR_INIT(normal_ooc_colour, "#002eb8")
...
+			to_chat(src, "<span class='danger'>Observers cannot use OOC pre-game.</span>")
+			return
 		if(!GLOB.dooc_allowed && (mob.stat == DEAD))
 			to_chat(usr, "<span class='danger'>OOC for dead mobs has been turned off.</span>")
 			return
```
Comment (PowerfulBacon) [inline code/modules/client/verbs/ooc.dm]:
```suggestion
		if(mob.stat == DEAD)
			if(SSticker.current_state < GAME_STATE_PLAYING)
				to_chat(src, "<span class='danger'>Observers cannot use OOC pre-game.</span>")
				return
			if(!GLOB.dooc_allowed)
				to_chat(usr, "<span class='danger'>OOC for dead mobs has been turned off.</span>")
				return
```

**Review by Archanial — APPROVED** [review]

**Comment by BNGP0** [comment]
Well revealing cogcrabs is obviously metagame, but i don't usually see people do it. It is not even profitable for them, since they can play as those cogcrabs and it is better for them to not reveal this info to the crew. But the other ghost spawns are not that vital to the round. Actually, i don't want to roll 20 fucking rounds to play as an ashwalker to see how no one is born in the tribe, so why not to keep at least some way of knowing if there are ghost roles i wanted to play before i join/observe? Even when i said we had ash walker role before the round, miners did not give a shit and we didn't get raided. Also, people don't tend to reveal lavaland syndie base in ooc, because if they do, miners will try to find and raid it roundstart.

**Pull request merged**